fms_dgt/base/block.py

```diff
@@ -0,0 +1,144 @@
+# Standard
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterable, List, Optional, Union
+
+# Third Party
+from datasets import Dataset
+import pandas as pd
+
+DATASET_ROW_TYPE = Union[Dict, pd.Series]
+DATASET_TYPE = Union[Iterable[DATASET_ROW_TYPE], pd.DataFrame, Dataset]
+
+
+class BaseBlock(ABC):
+    """Base Class for all Blocks"""
+
+    def __init__(
+        self,
+        name: str = None,
+        arg_fields: List[str] = None,
+        kwarg_fields: List[str] = None,
+        result_field: str = None,
+    ) -> None:
+
+        if not (isinstance(arg_fields, list) or arg_fields is None):
+            raise TypeError(f"arg_fields must be of type 'list'")
+        if not (isinstance(kwarg_fields, list) or kwarg_fields is None):
+            raise TypeError(f"kwarg_fields must be of type 'list'")
+        if not (isinstance(result_field, str) or result_field is None):
+            raise TypeError(f"result_field must be of type 'str'")
+
+        self._name = name
+
+        self._arg_fields = arg_fields
+        self._kwarg_fields = kwarg_fields
+        self._result_field = result_field
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def arg_fields(self):
+        return self._arg_fields
+
+    @property
+    def kwarg_fields(self):
+        return self._kwarg_fields
+
+    @property
+    def result_field(self):
+        return self._result_field
+
+    def get_args_kwargs(
+        self,
+        inp: BLOCK_ROW_TYPE,
+        arg_fields: Optional[List[str]] = None,
+        kwarg_fields: Optional[List[str]] = None,
+    ):
+
+        arg_fields = arg_fields or self.arg_fields or []
+        kwarg_fields = kwarg_fields or self.kwarg_fields or []
+
+        if isinstance(inp, (dict, pd.DataFrame, Dataset)):
+            args = [inp.get(arg) for arg in arg_fields]
+            kwargs = {kwarg: inp.get(kwarg) for kwarg in kwarg_fields}
+        else:
+            raise TypeError(f"Unexpected input type: {type(inp)}")
+
+        return args, kwargs
```

Comment:
Total code golf NIT, but...

```suggestion
        if isinstance(inp, (dict, pd.DataFrame, Dataset)):
            return (
                [inp.get(arg) for arg in arg_fields],
                {kwarg: inp.get(kwarg) for kwarg in kwarg_fields}
            )
        raise TypeError(f"Unexpected input type: {type(inp)}")

```

Reply:
ha, love it